re_view/src/re_view/core.cljs

```diff
@@ -54,21 +54,24 @@
           (set! (.-re$dbPatterns this) patterns))
         value))))
 
-(def ^:private kmap
+(defn- class-key
   "Mapping of methods-map keys to React lifecycle keys."
-  {:constructor "constructor"
-   :view/initial-state "$initialState"
-   :view/state "$state"
-   :view/did-catch "componentDidCatch"
-   :view/will-mount "componentWillMount"
-   :view/did-mount "componentDidMount"
-   :view/will-receive-props "componentWillReceiveProps"
-   :view/will-receive-state "componentWillReceiveState"
-   :view/should-update "shouldComponentUpdate"
-   :view/will-update "componentWillUpdate"
-   :view/did-update "componentDidUpdate"
-   :view/will-unmount "componentWillUnmount"
-   :view/render "render"})
+  [k]
+  (case k
+    :constructor "constructor"
+    :view/initial-state "$initialState"
+    :view/state "$state"
```

Comment:
ha, I found a bug (of my own making). this should be `"re$state"`. I can make this change after merging. I will probably also add `"re$props"`, as :view/state and :view/props are so commonly used and there is no point in running them through string manipulations, better to have direct lookup.